pandas/tests/test_generic.py

```diff
@@ -527,6 +529,21 @@ def test_sample(self):
         assert_panel_equal(p.sample(n=3, random_state=42), p.sample(n=3, axis=1, random_state=42))
         assert_frame_equal(df.sample(n=3, random_state=42), df.sample(n=3, axis=0, random_state=42))
 
+        # Test that function aligns weights with frame
+        df = DataFrame({'col1':[5,6,7], 'col2':['a','b','c'], }, index = [9,5,3])
+        s = Series([1,0,0], index=[3,5,9])
+        assert_frame_equal(df.loc[[3]], df.sample(1, weights=s))
```

Comment:
test also where you have a missing indexer (so should be nan filled), e.g. ``Series([1,0,0],index=[3,5,10])``